### PR TITLE
KAZOO-4467: remove ecallmgr CCV duplicates in AMQP messages

### PR DESCRIPTION
**Pull request by icehess**
Refactor FS command module and remove custom sip headers from channel

### CODE REVIEW

**Comment by lazedo** [comment]
@icehess needs rebase

File: applications/ecallmgr/src/ecallmgr_fs_route.erl [hunk 1]
```diff
@@ -282,7 +282,7 @@ search_for_route(Section, Node, FetchId, CallId, Props) ->
             ToSet = [{<<"Context">>, hunt_context(Props)}] ++ remote_sdp(Props),
             maybe_wait_for_authz(Section, Node, FetchId, CallId
                                  ,wh_json:set_values(ToSet, JObj)
-                                )
+                                 , Props)
```
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_route.erl]: @icehess why are you passing the large Props down ? you can call ecallmgr_uitil:custom_sip_headers(Props) and add it to the ToSet var above, then extract in ecallmgr_fs_xml.

File: applications/ecallmgr/src/ecallmgr_fs_route.erl [hunk 2]
```diff
@@ -296,17 +296,17 @@ remote_sdp(Props) ->
         RemoteSDP -> [{<<"Remote-SDP">>, RemoteSDP}]
     end.
 
--spec maybe_wait_for_authz(atom(), atom(), ne_binary(), ne_binary(), wh_json:object()) -> 'ok'.
-maybe_wait_for_authz(Section, Node, FetchId, CallId, JObj) ->
-    case ecallmgr_config:is_true(<<"authz_enabled">>, 'false')
+-spec maybe_wait_for_authz(atom(), atom(), ne_binary(), ne_binary(), wh_json:object(), wh_proplist()) -> 'ok'.
+maybe_wait_for_authz(Section, Node, FetchId, CallId, JObj, Props) ->
+    case wh_util:is_true(ecallmgr_config:get(<<"authz_enabled">>, 'false'))
```
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_route.erl]: why this change ? doesn't `ecallmgr_config:is_true` work ?

File: applications/ecallmgr/src/ecallmgr_fs_xml.erl [hunk 3]
```diff
@@ -590,10 +595,20 @@ get_channel_params_fold(Key, Val) ->
...
     wh_json:to_proplist(wh_json:get_value(<<"Custom-SIP-Headers">>, JObj, wh_json:new())).
 
+-spec normalize_custom_sip_header_name(any()) -> any().
+normalize_custom_sip_header_name({<<"variable_", K/binary>>, V}) -> {K, V};
+normalize_custom_sip_header_name(A) -> A.
+
+-spec is_custom_sip_header(any()) -> boolean().
+is_custom_sip_header({<<"variable_sip_h_X-", _/binary>>, _}) -> 'true';
+is_custom_sip_header(_) -> 'false'.
+
```
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_xml.erl]: this seems like a duplication of the functions in `ecallmgr_util`

Comment (icehess) [inline applications/ecallmgr/src/ecallmgr_fs_xml.erl]: yes indeed it's a duplicate of those functions. Because I need headers to be prefixed with `sip_h_` but `ecallmgr_util` function is using `normalize_custom_sip_header_name` to normalize them. With the discussion that I had with @k-anderson, he seems to happy with the clone version.

Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_xml.erl]: you wouldn't need all of these if you follow the suggestion of not propagating the large Props

File: applications/ecallmgr/src/ecallmgr_fs_route.erl (hunk 2 above)
Comment (icehess) [inline applications/ecallmgr/src/ecallmgr_fs_route.erl]: It messed up during the rebase, I'll fix this

File: applications/ecallmgr/src/ecallmgr_fs_route.erl (hunk 1 above)
Comment (icehess) [inline applications/ecallmgr/src/ecallmgr_fs_route.erl]: I need the headers to be prefixed with `sip_h_` so what about bring my `get_custom_sip_headers` function from `fs_xml` to here and get all custom variables and passed them down instead of the whole Props?

File: applications/ecallmgr/src/ecallmgr_fs_xml.erl [hunk 4]
```diff
@@ -406,19 +411,18 @@ route_resp_pre_park_action(JObj) ->
...
--spec check_dtmf_type(wh_json:object()) -> 'undefined' | xml_el().
-check_dtmf_type(JObj) ->
-    SDPRemote = wh_json:get_value(<<"Remote-SDP">>, JObj, <<"101 telephone-event">>),
-    case binary:match(SDPRemote, <<"101 telephone-event">>) of
-        'nomatch' -> action_el(<<"start_dtmf">>);
-        _ -> 'undefined'
+-spec check_dtmf_type(wh_proplist()) -> 'undefined' | xml_el().
+check_dtmf_type(Props) ->
+    case props:get_value(<<"variable_switch_r_sdp">>, Props, <<"101 telephone-event">>) of
+        <<"101 telephone-event">> -> io:format("undefined ~n~n"),'undefined';
```
Comment (k-anderson) [inline applications/ecallmgr/src/ecallmgr_fs_xml.erl]: @icehess please remove io:format("undefined ~n~n")
